script-nivel-3.js

```diff
@@ -0,0 +1,38 @@
+import { CAR_DATA } from "./script-nivel-3-data.js";
+
+displaySelectedCars(selectCarByPrice(CAR_DATA));
+
+
+function selectCarByPrice(data) {
+    const selectedCars = data;
+    for (let i = 0; i < data.length; i++) {
+        // selectedCars[i] = {"nume" : data[i]["nume"]};
+        for (let j = 0; j < data[i]["masini"].length; j++) {
+            if (data[i]["masini"][j]["pret"] > 100) {
+                selectedCars[i]["masini"][j] = {};
+            }
+        }
+    }
+    return selectedCars;
+}
+
+
+function displaySelectedCars(cars) {
+    for (let i = 0; i < cars.length; i++) {
+        document.getElementById("result").innerHTML += cars[i]["nume"] + "<br>";
+        
+        let contor = 0;
+        for (let j = 0; j < cars[i]["masini"].length; j++) {
```

Comment:
un obiect json precum cars[i] are doua moduri de selectie cel specific python in care cautam dupa un string(sau dictionary key search) 
```js
cars[i]["masini"]
```si 
mai avem si property key search 
```js
cars[i].masini
```
proprietate ce o putem folosi combinata cu destructurarea obiectelor pentru a extrage variabilele de interes din obiect, pentru a avea codul mai simplist si pentru a putea folosi functiile de iterare precum map, forEach si filter
```js
const {masini} = cars[i];
const carsBelowPrice = masini.filter(masina => masina.pret < 100).length;
```

Tendinta de asemenea propune sa nu mai folosim for loops, din motivul ca sunt mai greu de citit, si iteratoarele map, filter samd. pot exploata procesarea paralela superior pentru seturi mari de date